Core/Inc/CAN/CAN_IDs.h

```diff
@@ -9,198 +9,92 @@
+    CAN_GENERAL_LIST_DATA(ID_LOST_MSG_L_MOTOR, 112, 0)                                   \
+    CAN_GENERAL_LIST_DATA(ID_BUS_OFF_L_MOTOR, 112, 1)                                    \
+    CAN_GENERAL_LIST_DATA(ID_CAN_STATE_L_MOTOR, 112, 2)                                  \
+    CAN_GENERAL_LIST_DATA(ID_INV_STATE_L_MOTOR, 113, 0)                                  \
+    CAN_GENERAL_LIST_DATA(ID_FAILURE_L_MOTOR, 113, 1)                                    \
+    CAN_GENERAL_LIST_DATA(ID_ALARM_L_MOTOR, 113, 2)                                      \
+    CAN_GENERAL_LIST_DATA(ID_SPEED_R_MOTOR, 120, 0)                                      \
+    CAN_GENERAL_LIST_DATA(ID_TORQUE_R_MOTOR, 120, 1)                                     \
+    CAN_GENERAL_LIST_DATA(ID_POWER_R_MOTOR, 120, 2)                                      \
+    CAN_GENERAL_LIST_DATA(ID_CURRENT_R_MOTOR, 120, 3)                                    \
+    CAN_GENERAL_LIST_DATA(ID_ENERGY_R_MOTOR, 121, 0)                                     \
+    CAN_GENERAL_LIST_DATA(ID_OVERLOAD_R_MOTOR, 121, 1)                                   \
+    CAN_GENERAL_LIST_DATA(ID_TEMPERATURE1_R, 121, 2)                                     \
+    CAN_GENERAL_LIST_DATA(ID_TEMPERATURE2_R, 121, 3)                                     \
+    CAN_GENERAL_LIST_DATA(ID_LOST_MSG_R_MOTOR, 122, 0)                                   \
+    CAN_GENERAL_LIST_DATA(ID_BUS_OFF_R_MOTOR, 122, 1)                                    \
+    CAN_GENERAL_LIST_DATA(ID_CAN_STATE_R_MOTOR, 122, 2)                                  \
+    CAN_GENERAL_LIST_DATA(ID_INV_STATE_R_MOTOR, 123, 0)                                  \
+    CAN_GENERAL_LIST_DATA(ID_FAILURE_R_MOTOR, 123, 1)                                    \
+    CAN_GENERAL_LIST_DATA(ID_ALARM_R_MOTOR, 123, 2)                                      \
+    CAN_GENERAL_LIST_DATA(ID_PANEL_DEBUG_1, 351, 0)                                      \
+    CAN_GENERAL_LIST_DATA(ID_PANEL_DEBUG_2, 351, 1)                                      \
+    CAN_GENERAL_LIST_DATA(ID_PANEL_DEBUG_3, 351, 2)                                      \
+    CAN_GENERAL_LIST_DATA(ID_PANEL_DEBUG_4, 351, 3)
 
-#define CAN_ID_QUAN 54
+typedef enum {
+    RESERVED_GENERAL_ID_0,
+#define CAN_GENERAL_LIST_DATA(var_name, msg_id, msg_wrd) var_name,
+    VARIABLES_GENERAL_CAN_TX
+#undef CAN_GENERAL_LIST_DATA
+        CAN_GENERAL_ID_QUAN
+} CAN_general_list_e;
 
 typedef struct {
+    CAN_general_list_e var;
     uint16_t id;
     uint8_t pos;
-} CAN_ID_t;
+} CAN_var_inf;
 
-extern CAN_ID_t CAN_ID_map[CAN_ID_QUAN];
-// TODO(renanmoreira): IMPLEMENT X-MACROS FOR CAN_ID'S
-#define ECU_CAN_INITIAL_ID        100
-#define ECU_CAN_FINAL_ID          149
-#define ECU_CAN_LAST_POPULATED_ID 113
-#define ECU_CAN_FIRST_DEBUG_ID    351
-#define ECU_CAN_LAST_DEBUG_ID     351
+typedef struct {
+    uint16_t external_ID;
+    int16_t pos[4];
+} datalog_send_t;
```

Comment:
Nesse pos fica o id da mensagem que vai ser enviada, certo? Se sim, deixa isso mais claro.
```suggestion
typedef struct {
    uint16_t external_ID;
    CAN_general_list_e message_to_send[4];
} datalog_send_t;
```